library/Garden/TwigTrait.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * @author Alexander Kim <alexander.k@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+namespace Garden;
+
+/**
+ * For classes that need to cache some static values and configs.
+ *
+ */
+trait TwigTrait {
```

Comment:
Could name of the trait and its properties be less specific to Twig? I'm not altogether opposed to hardcoding the usage of Twig in the body of these methods, but the next time Vanilla switches templating engines, it'd be nice if we we could swap out the guts of these methods without the need to also rename traits and methods for every class using this trait.

Reply:
IMHO  that is the only reason for this trait to exist. The idea is to have multiple traits specific to any particular library we need and to have common entry point for other classes to use it. So whenever any class need Twig to get - it uses TwigTrait, if class needs Mustache - we can have MustacheTrait to implement if we need Smarty - we can have SmartyTrait.

In this particular scenario my idea is to have method MustacheInit, SmartyInit etc. And our new knowledge base controller 
https://github.com/vanilla/knowledge/blob/19610933abbdc208b78a2b072882ffe82644c0a6/plugins/knowledge/Controllers/PageController.php#L40
will call particular templating system/library to instantiate it for particular Controller. Where every controller will define itself which templating it need by `use TwigTrait` or `use SmartyTrait` etc.